db.go

```diff
@@ -134,6 +163,11 @@ type TxListReply struct {
 	Err     error
 }
 
+// BlockAddrIndex represents the indexing structure for addresses.
+// It maps a hash160 to a list of transaction locations within a block that
+// either pays to or spends from the passed UTXO for the hash160.
+type BlockAddrIndex map[[ripemd160.Size]byte][]*btcwire.TxLoc
```

Comment:
I like @jrick's suggestion of adding a new type to `btcutil`, creates a useful type for potential future use. 

What about this for P2SH?
```go
type ScriptHash [ripemd160.Size]byte
```

Yeah the index also contains P2SH addresses as well. On the indexing level, all Script data pushes a certain size in a scriptPubKey are indexed to that tx. While the query level is strictly "address based" atm.